Modules/Quotas/src/Quotas.Application/IntegrationEvents/Incoming/TierOfIdentityChanged/TierOfIdentityChangedIntegrationEventHandler.cs

```diff
@@ -0,0 +1,31 @@
+﻿using Backbone.Modules.Quotas.Application.Infrastructure.Persistence.Repository;
+using Backbone.Modules.Quotas.Application.Metrics;
+using Enmeshed.BuildingBlocks.Application.Abstractions.Infrastructure.EventBus;
+
+namespace Backbone.Modules.Quotas.Application.IntegrationEvents.Incoming.TierOfIdentityChanged;
+public class TierOfIdentityChangedIntegrationEventHandler : IIntegrationEventHandler<TierOfIdentityChangedIntegrationEvent>
+{
+    private readonly IIdentitiesRepository _identitiesRepository;
+    private readonly ITiersRepository _tiersRepository;
+    private readonly IMetricStatusesService _metricStatusesService;
+
+    public TierOfIdentityChangedIntegrationEventHandler(IIdentitiesRepository identitiesRepository, ITiersRepository tiersRepository, IMetricStatusesService metricStatusesService)
+    {
+        _identitiesRepository = identitiesRepository;
+        _tiersRepository = tiersRepository;
+        _metricStatusesService = metricStatusesService;
+    }
+
+    public async Task Handle(TierOfIdentityChangedIntegrationEvent @event)
+    {
+        var identity = await _identitiesRepository.Find(@event.IdentityAddress, CancellationToken.None, track: true);
+        var newTier = await _tiersRepository.Find(@event.NewTier, CancellationToken.None, track: true);
+        identity.ChangeTier(newTier);
```

Comment:
I know you implemented it as I requested, but I have a suggestion:

How about this:

```csharp
await identity.ChangeTier(newTier, metricCalculatorFactory, CancellationToken.None);
```
?

The Identity should then automatically recalculate the MetricStatuses (without the MetricStatusesService)

Reply:
Did you mean `metricCalculatorFactory` or `metricStatusesService`?

Reply:
I believe I interpreted that correctly.